openmc/statepoint.py

```diff
@@ -478,21 +418,20 @@ def tally_derivatives(self):
 
                 # Create each derivative object and add it to the dictionary.
                 for d_id in deriv_ids:
-                    base = 'tallies/derivatives/derivative {:d}'.format(d_id)
+                    group = self._f['tallies/derivatives/derivative {}'
+                                    .format(d_id)]
                     deriv = openmc.TallyDerivative(derivative_id=d_id)
-                    deriv.variable = \
-                         self._f[base + '/independent variable'].value.decode()
+                    deriv.variable = group['independent variable'].value.decode()
                     if deriv.variable == 'density':
-                        deriv.material = self._f[base + '/material'].value
+                        deriv.material = group['material'].value
                     elif deriv.variable == 'nuclide_density':
-                        deriv.material = self._f[base + '/material'].value
-                        deriv.nuclide = \
-                             self._f[base + '/nuclide'].value.decode()
+                        deriv.material = group['material'].value
+                        deriv.nuclide = group['nuclide'].value.decode()
                     elif deriv.variable == 'temperature':
-                        deriv.material = self._f[base + '/material'].value
+                        deriv.material = group['material'].value
                     else:
-                        raise RuntimeError('Unrecognized tally differential '
-                                           'variable')
+                        raise ValueError('Unrecognized tally differential '
```

Comment:
Actually, the setter for `TallyDerivative.variable` already handles this so I'm just going to remove this exception.